src/dlgRoomSymbol.cpp

```diff
@@ -0,0 +1,219 @@
+#endif
+    if (symbolCountsList.size() > 1) {
+        std::sort(symbolCountsList.begin(), symbolCountsList.end());
+    }
+    QStringList displayStrings;
+    for (int i = symbolCountsList.size() - 1; i >= 0; --i) {
+        itSymbolUsed.toFront();
+        while (itSymbolUsed.hasNext()) {
+            itSymbolUsed.next();
+            if (itSymbolUsed.value() == symbolCountsList.at(i)) {
+                displayStrings.append(tr("%1 {count:%2}",
+                                         // Intentional comment to separate arguments
+                                         "Everything after the first parameter (the '%1') will be removed by processing "
+                                         "it as a QRegularExpression programmatically, ensure the translated text has "
+                                         "` {` immediately after the '%1', and '}' as the very last character, so that the "
+                                         "right portion can be extracted if the user clicks on this item when it is shown "
+                                         "in the QComboBox it is put in.")
+                                              .arg(itSymbolUsed.key())
+                                              .arg(QString::number(itSymbolUsed.value())));
+            }
+        }
+    }
+    return displayStrings;
+}
+
+void dlgRoomSymbol::accept()
+{
+    QDialog::accept();
+    emit signal_save_symbol(getNewSymbol(), selectedColor, mpRooms);
+}
+
+QString dlgRoomSymbol::getNewSymbol()
+{
+    if (mpSymbols.size() <= 1) {
+        return lineEdit_roomSymbol->text();
+    } else {
+        QRegularExpression countStripper(QStringLiteral("^(.*) {.*}$"));
+        QRegularExpressionMatch match = countStripper.match(comboBox_roomSymbol->currentText());
+        if (match.hasMatch() && match.lastCapturedIndex() > 0) {
+            // captured(0) is the whole string that matched, which is
+            // not what we want:
+            return match.captured(1);
+        }
+        return comboBox_roomSymbol->currentText();
+    }
+}
+
+void dlgRoomSymbol::updatePreview()
+{
+    auto realColor = selectedColor != nullptr ? selectedColor : defaultColor();
+    label_preview->setText(getNewSymbol());
+    auto bgStyle = QStringLiteral("background-color: %1; border: 1px solid; border-radius: 1px;").arg(realColor.name());
+    pushButton_roomSymbolColor->setStyleSheet(bgStyle);
+    label_preview->setStyleSheet(
+            QStringLiteral("color: %1; background-color: %2; border: %3;")
+                    .arg(realColor.name(), roomColor.name(), mpHost->mMapperShowRoomBorders ? QStringLiteral("1px solid %1").arg(mpHost->mRoomBorderColor.name()) : QStringLiteral("none")));
+}
+
+void dlgRoomSymbol::openColorSelector()
+{
+    auto* dialog = selectedColor != nullptr ? new QColorDialog(selectedColor, this) : new QColorDialog(defaultColor(), this);
+    dialog->setWindowTitle(tr("Pick color"));
+    dialog->open(this, SLOT(colorSelected(const QColor&)));
```

Comment:
I think you need to set `Qt::WA_DeleteOnClose` on this one as well. Test by opening a ton of dialogs and closing them, see how the memory use is like.

(FTR some people run Mudlet for months on end without restarting...)